default.nix

```
@@ -1,5 +1,23 @@
 { # The files would be going to ~/.config/doom (~/.doom.d)
   doomPrivateDir
+  /* A Doom configuration directory from which to build the Emacs package environment.
+
+     Can be used, for instance, to prevent rebuilding the Emacs environment
+     each time the `config.el` changes.
+
+     Can be provided as a directory or derivation. If not given, package
+     environment is built against `doomPrivateDir`.
+
+     Example:
+       doomPackageDir = pkgs.linkFarm "my-doom-packages" [
+         # straight needs a (possibly empty) `config.el` file to build
+         { name = "config.el"; path = pkgs.emptyFile; }
+         { name = "init.el"; path = ./doom.d/init.el; }
+         { name = "packages.el"; path = pkgs.writeText "(package! inheritenv)"; }
+         { name = "modules"; path = ./my-doom-module; }
+       ];
+   */
+,  doomPackageDir ? null
```

Actually this could also be expressed as:

```suggestion
,  doomPackageDir ? doomPrivateDir
```

---

Great call, done.